### PR TITLE
The cnetdiff application has been refactored to be callable; Makefile tests have been removed and replaced by gtests.

### PR DESCRIPTION
**Pull request by kledmundson**
## Description

The cnetdiff application has been refactored to be callable; Makefile tests have been removed and replaced by gtests.

## Related Issue

https://github.com/DOI-USGS/ISIS3/issues/5322

## How Has This Been Validated?

ISIS make file tests have been replaced with the following gtests.

CnetdiffReportFirst
CnetdiffReportFull

Results of ctest suite...

ctest -R Cnetdiff --output-on-failure
Test project
    Start 1798:CompareNetsReportFirstDiff.FunctionalTestCnetdiffReportFirst
1/2 Test #1798: CompareNetsReportFirstDiff.FunctionalTestCnetdiffReportFirst ...   Passed    0.74 sec
Start 1799: CompareNetsReportFull.FunctionalTestCnetdiffReportFull
2/2 Test #1799:
CompareNetsReportFull.FunctionalTestCnetdiffReportFull .........   Passed    0.56 sec

100% tests passed, 0 tests failed out of 2
Total Test time (real) =   1.50 sec

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ release.
   outputsuffix, both, or neither for naming convention purposes. [#5162](https://github.com/DOI-USGS/ISIS3/pull/5162)
 
 ### Added
-- Added rclone to run dependencies in meta.yaml [#5183](https://github.com/DOI-USGS/ISIS3/issues/5183)
+- Cnetedit has been refactored to be callable; old Makefile tests have been removed and replaced by gtests. Issue: [#5322](https://github.com/USGS-Astrogeology/ISIS3/issues/5322),
```
Comment (acpaquette) [inline CHANGELOG.md]: Can this be moved to changed? If there is no `Changed` section feel free to add a new header titled `Changed`

File: isis/src/control/apps/cnetdiff/cnetdiff.cpp [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+      PvlGroup differences("Results");
+      if (filesMatch) {
+        differences += PvlKeyword("Compare", "Identical");
+      }
+      else {
+        differences += PvlKeyword("Compare", "Different");
+        differences += PvlKeyword("Reason", differenceReason);
+      }
+
+      log.addGroup(differences);
```
Comment (acpaquette) [inline isis/src/control/apps/cnetdiff/cnetdiff.cpp]: Could any addition to the log be changed to `log.addLogGroup`. This function logs to the app at the same time as the group is added to the keyword. With that you can remove the logging loop in the application `main.cpp`

File: isis/src/control/objs/ControlNetDiff/ControlNetDiff.cpp [hunk 3]
```diff
@@ -88,14 +87,31 @@ namespace Isis {
...
+   * differences.
+   *
+   * @param cnv1 first Control Network
+   * @param cnv2 second Control Network
+   * @param net1Name FileName of the first Control Network
+   * @param net2Name FileName of the second Control Network
+   * @return Pvl The collection of all differences
+   */
+  Pvl ControlNetDiff::compare(ControlNetVersioner &cnv1, ControlNetVersioner &cnv2,
+                              FileName &net1Name, FileName &net2Name) {
```
Comment (acpaquette) [inline isis/src/control/objs/ControlNetDiff/ControlNetDiff.cpp]: I don't really understand the necessity for this change. What does exposing the new function do that the old signature prevented?

**Review by acpaquette — COMMENTED** [review]
Overall looks great! Couple small things to address

File: CHANGELOG.md (hunk 1 above)
Comment (kledmundson) [reply CHANGELOG.md]: Done.

File: isis/src/control/apps/cnetdiff/cnetdiff.cpp (hunk 2 above)
Comment (kledmundson) [reply isis/src/control/apps/cnetdiff/cnetdiff.cpp]: Also done.

File: isis/src/control/objs/ControlNetDiff/ControlNetDiff.cpp (hunk 3 above)
Comment (kledmundson) [reply isis/src/control/objs/ControlNetDiff/ControlNetDiff.cpp]: You're right. Major brain lock on my part. Reverted back to the original ControlNetDiff.cpp;h.

**Comment by kledmundson** [comment]
Thanks for the very quick review Adam.
I've made the changes and have hopefully addressed your suggestions.

Ken

File: CHANGELOG.md [hunk 4]
```diff
@@ -45,7 +46,6 @@ release.
   outputsuffix, both, or neither for naming convention purposes. [#5162](https://github.com/DOI-USGS/ISIS3/pull/5162)
 
 ### Added
-- Added rclone to run dependencies in meta.yaml [#5183](https://github.com/DOI-USGS/ISIS3/issues/5183)
```
Comment (acpaquette) [inline CHANGELOG.md]: @kledmundson One last thing, it looks like this entry got erroneously removed. Could we add it back in?

Comment (kledmundson) [reply CHANGELOG.md]: Done.

**Review by acpaquette — APPROVED** [review]

**Pull request merged**